### PR TITLE
Added unload_library method to shared_library

### PR DESCRIPTION
**Pull request by ahcorde**
As discussed in the PR https://github.com/ros/class_loader/pull/139#pullrequestreview-387397419 I will include a unload method to the sharedLibrary abstraction

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Have you tried this patch? I would assume if you invoke `unload_library` first that the constructor will fail since it will try to unload the library again.

File: src/shared_library.cpp [hunk 1]
```diff
@@ -40,11 +40,32 @@ SharedLibrary::SharedLibrary(const std::string & library_path)
...
+  } else {
+    if (lib.library_path != NULL) {
+      if (rcutils_allocator_is_valid(&lib.allocator)) {
+        lib.allocator.deallocate(lib.library_path, lib.allocator.state);
+        lib.allocator = rcutils_get_zero_initialized_allocator();
+      } else {
+        std::cerr << "ShareLibrary not able to deallocate memory on constructor" << std::endl;
+      }
+    }
+  }
```
Comment (dirk-thomas) [inline src/shared_library.cpp:59]: Why does this logic need to be here? Can't this call the unload API when a certain condition is met instead? Having knowledge about how the stuff needs to be deallocated etc. seems to break encapsulation.

Comment (ahcorde) [reply src/shared_library.cpp:50]: fixed https://github.com/ros2/rcpputils/pull/50/commits/cb11c673b72e18723647486e19bce1c06dcace72

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9983/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5583/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8122/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9858/)

**Comment by ahcorde** [comment]
Openning and clossing the issue to retest the github actions

**Comment by ahcorde** [comment]
Openning and clossing the issue to retest the github actions

**Pull request merged**